ui/imports/shared/views/RecipientView.qml

```diff
@@ -0,0 +1,236 @@
+import QtQuick 2.13
+import QtQuick.Layouts 1.13
+
+import StatusQ.Controls 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1 as StatusQUtils
+
+import AppLayouts.Wallet 1.0
+
+import utils 1.0
+
+import "../controls"
+
+Loader {
+    id: root
+
+    property var store
+    property bool isBridgeTx: false
+    property bool interactive: true
+    property bool showUnpreferredNetworks: false
+    property var selectedAsset
+    property var selectedRecipient: null
+    property int selectedRecipientType
+
+    readonly property bool ready: (d.isAddressValid || root.isENSValid) && !d.isPending
+    property string addressText
+    property bool isENSValid: false
+    property string resolvedENSAddress
+
+    signal recalculateRoutesAndFees()
+    signal isLoading()
+
+    onAddressTextChanged: d.isPending = false
+
+    onSelectedRecipientChanged: {
+        root.isLoading()
+        d.waitTimer.restart()
+        if(!!root.selectedRecipient && root.selectedRecipientType !== TabAddressSelectorView.Type.None) {
+            switch(root.selectedRecipientType) {
+            case TabAddressSelectorView.Type.SavedAddress: {
+                if (root.selectedRecipient.ens.length > 0) {
+                    d.isPending = true
+                    return Qt.callLater(d.resolveENS,  root.selectedRecipient.ens)
```

Comment:
@Khushboo-dev-cpp I don't see that we will benefit from `Qt.callLater` in this case. You're free to remove it.